api_v3/services/CategoryService.php

```diff
@@ -316,12 +316,19 @@ function indexAction($id, $shouldUpdate = true)
 		return $categoryDb->getId();
 	}
 	
-	private function handleCoreException(kCoreException $ex, category $categoryDb, KalturaCategory $category)
+	private function handleCoreException(kCoreException $ex, category $categoryDb, KalturaCategory $category, $originalCategoryDb = null)
 	{
 		switch($ex->getCode())
 		{
 			case kCoreException::DUPLICATE_CATEGORY:
-				throw new KalturaAPIException(KalturaErrors::DUPLICATE_CATEGORY, $categoryDb->getFullName());
+				if (isset($originalCategoryDb))
```

Comment:
better use something like
$name =  $categoryDb->getFullName();
if isset(originalCategoryDb)
     $name =
and then have one throw